matchbox_socket/src/webrtc_socket/native.rs

```diff
@@ -188,110 +187,113 @@ impl Messenger for NativeMessenger {
+                .await
+                .unwrap();
+
+            let (data_channel_ready_txs, data_channels_ready_fut) =
+                create_data_channels_ready_fut(config);
+
+            let data_channels = create_data_channels(
+                &connection,
+                data_channel_ready_txs,
+                signal_peer.id,
+                peer_disconnected_tx.clone(),
+                messages_from_peers_tx,
+                &config.channels,
+            )
+            .await;
+
+            let offer = loop {
+                match peer_signal_rx.next().await.expect("error") {
+                    PeerSignal::Offer(offer) => {
+                        break offer;
+                    }
+                    _ => {
+                        warn!("ignoring other signal!!!");
+                    }
                 }
+            };
+            debug!("received offer");
+            let remote_description = RTCSessionDescription::offer(offer).unwrap();
+            connection
+                .set_remote_description(remote_description)
+                .await
+                .unwrap();
+
+            let answer = connection.create_answer(None).await.unwrap();
+            signal_peer.send(PeerSignal::Answer(answer.sdp.clone()));
+            connection.set_local_description(answer).await.unwrap();
+
+            let trickle_fut = complete_handshake(
+                trickle,
+                &connection,
+                peer_signal_rx,
+                data_channels_ready_fut,
+            )
+            .await;
+
+            HandshakeResult::<Self::DataChannel, Self::HandshakeMeta> {
+                peer_id: signal_peer.id,
+                data_channels: to_peer_message_tx,
+                metadata: (
+                    to_peer_message_rx,
+                    data_channels,
+                    trickle_fut,
+                    peer_disconnected_rx,
+                ),
             }
-        };
-        debug!("received offer");
-        let remote_description = RTCSessionDescription::offer(offer).unwrap();
-        connection
-            .set_remote_description(remote_description)
-            .await
-            .unwrap();
-
-        let answer = connection.create_answer(None).compat().await.unwrap();
-        signal_peer.send(PeerSignal::Answer(answer.sdp.clone()));
-        connection
-            .set_local_description(answer)
-            .compat()
-            .await
-            .unwrap();
-
-        let trickle_fut = complete_handshake(
-            trickle,
-            &connection,
-            peer_signal_rx,
-            data_channels_ready_fut,
-        )
-        .await;
-
-        HandshakeResult {
-            peer_id: signal_peer.id,
-            data_channels: to_peer_message_tx,
-            metadata: (
-                to_peer_message_rx,
-                data_channels,
-                trickle_fut,
-                peer_disconnected_rx,
-            ),
         }
+        .compat()
```

Comment:
Can we add a comment here saying .compat is critical for wasm<->native connections?